packages/select/src/MenuButton.tsx

```diff
@@ -0,0 +1,290 @@
+} & Required<
+  Pick<
+    JSX.IntrinsicElements['div'],
+    'aria-labelledby' | 'aria-controls' | 'aria-expanded' | 'aria-describedby'
+  >
+>;
+
+function MenuButton({
+  children,
+  value,
+  text,
+  name,
+  deselected,
+  readOnly,
+  onDeselect,
+  onFocusFirstOption,
+  onFocusLastOption,
+  onClose,
+  onOpen,
+  refElement,
+  ...ariaProps
+}: Props) {
+  const { mode, size, open, disabled } = useContext(SelectContext);
+
+  const { usingKeyboard } = useUsingKeyboardContext();
+
+  const [isHovered, setIsHovered] = useState(false);
+
+  const colorSet = colorSets[mode];
+  const sizeSet = sizeSets[size];
+
+  const baseBoxShadow = `inset 0 -1px 0 0 ${colorSet.shadow.base}`;
+  let boxShadowExpanded = `0 0 0 3px ${colorSet.shadow.expanded}`;
+
+  if (mode === Mode.Dark) {
+    boxShadowExpanded = `inset 0 2px 2px 0 #373B3C, ${boxShadowExpanded}`;
+  }
+
+  const focusStyle = cx({
+    [css`
+      box-shadow: ${boxShadowExpanded};
+    `]: usingKeyboard,
+  });
+
+  const onMouseDown = useCallback(
+    (event: React.MouseEvent) => {
+      if (disabled) {
+        // prevent receiving focus
+        event.preventDefault();
+      }
+    },
+    [disabled],
+  );
+
+  const onKeyDown = useCallback(
+    (event: React.KeyboardEvent) => {
+      if (disabled) {
+        return;
+      }
+
+      /* istanbul ignore if */
+      if (event.ctrlKey || event.shiftKey || event.altKey) {
+        return;
+      }
+
+      let bubble = false;
+
+      switch (event.keyCode) {
+        case 9: // Tab
+          onClose();
+          bubble = true;
+          break;
+        case 27: // Escape
+          if (open) {
+            onClose();
+          } else {
+            onDeselect();
+          }
+          break;
+        case 38: // ArrowUp
+          onOpen();
+          onFocusLastOption();
+          break;
+        case 40: // ArrowDown
+          onOpen();
+          onFocusFirstOption();
+          break;
+        /* istanbul ignore next */
+        default:
+          bubble = true;
+      }
+
+      if (!bubble) {
+        event.preventDefault();
+        event.stopPropagation();
+      }
+    },
+    [
+      disabled,
+      onClose,
+      onDeselect,
+      onFocusFirstOption,
+      onFocusLastOption,
+      onOpen,
+      open,
+    ],
+  );
+
+  const onClick = useCallback(() => {
+    if (!disabled) {
+      if (open) {
+        onClose();
+      } else {
+        onOpen();
+      }
+      refElement.current!.focus();
+    }
+  }, [disabled, onClose, onOpen, open, refElement]);
+
+  const offHover = useCallback(() => {
+    setIsHovered(false);
+  }, []);
+
+  const onHover = useCallback(() => {
+    setIsHovered(true);
+  }, []);
+
+  // Only directly hovering over the button should induce hover
+  const preventChildrenFromTriggeringHoverProps = {
+    onMouseEnter: offHover,
+    onMouseLeave: onHover,
+  };
+
+  return (
+    <div
+      // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
```

Comment:
I'm seeing an error in the accessibility tab:
`Required ARIA child role not present: textbox`

Reply:
the warnings are still based on aria 1.1. in aria 1.2 a textbox is not required
https://www.w3.org/TR/wai-aria-1.2/#combobox

Reply:
Good to know!